lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1321,7 +1296,8 @@ def deploy_vm(self):
         self.configure_cpu_and_memory(vm_obj=vm_obj, vm_creation=True)
         self.configure_disks(vm_obj=vm_obj)
         self.configure_network(vm_obj=vm_obj)
-        if self.should_deploy_from_template() and len(self.params['customizations']) > 0:
+
+        if self.should_deploy_from_template() and len(self.params['customization']) > 0:
```

Comment:
Changed. I am going tired and don't even want to discuss anything anymore.

I'd rather get this pulled and then improved with individual PRs. I have rebased/merged conflicts 4 times now because fixes in this patch are being merged faster than this PR. So please merge this, and then offer PRs instead.